diff -urN kde-workspace-4.10.3/plasma/desktop/applets/pager/package/contents/ui/main.qml kde-workspace-4.10.3-patched/plasma/desktop/applets/pager/package/contents/ui/main.qml
--- kde-workspace-4.10.3/plasma/desktop/applets/pager/package/contents/ui/main.qml	2013-05-03 15:36:45.718117084 +1100
+++ kde-workspace-4.10.3-patched/plasma/desktop/applets/pager/package/contents/ui/main.qml	2013-05-13 17:37:18.851865667 +1100
@@ -154,7 +154,7 @@
                             pixmap: model.icon
                             height: nativeHeight
                             width: nativeWidth
-                            visible: pager.showWindowIcons && (windowRect.width >= icon.width) && (windowRect.height >= icon.height)
+                            visible: pager.showWindowIcons
                         }
 
                         MouseArea {
