diff -urN kde-workspace-4.11.3/kwin/libkdecorations/CMakeLists.txt kde-workspace-4.11.3-patched/kwin/libkdecorations/CMakeLists.txt
--- kde-workspace-4.11.3/kwin/libkdecorations/CMakeLists.txt	2013-08-29 03:59:38.242529479 +1100
+++ kde-workspace-4.11.3-patched/kwin/libkdecorations/CMakeLists.txt	2013-11-06 22:24:34.091169600 +1100
@@ -24,4 +24,5 @@
     kdecoration.h
     kdecorationfactory.h
     kcommondecoration.h
+    kdecorationbridge.h
     DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel )
diff -urN kde-workspace-4.11.3/kwin/libkdecorations/kdecorationbridge.h kde-workspace-4.11.3-patched/kwin/libkdecorations/kdecorationbridge.h
--- kde-workspace-4.11.3/kwin/libkdecorations/kdecorationbridge.h	2013-08-29 03:59:38.244529466 +1100
+++ kde-workspace-4.11.3-patched/kwin/libkdecorations/kdecorationbridge.h	2013-11-06 22:24:34.091169600 +1100
@@ -86,7 +86,7 @@
     virtual void grabXServer(bool grab) = 0;
 };
 
-class KDecorationBridgeUnstable
+class KWIN_EXPORT KDecorationBridgeUnstable
     : public KDecorationBridge
 {
 public:
diff -urN kde-workspace-4.11.3/kwin/libkdecorations/kdecoration.cpp kde-workspace-4.11.3-patched/kwin/libkdecorations/kdecoration.cpp
--- kde-workspace-4.11.3/kwin/libkdecorations/kdecoration.cpp	2013-11-02 02:46:22.586972614 +1100
+++ kde-workspace-4.11.3-patched/kwin/libkdecorations/kdecoration.cpp	2013-11-06 22:28:11.603234798 +1100
@@ -434,6 +434,7 @@
 KDecorationUnstable::KDecorationUnstable(KDecorationBridge* bridge, KDecorationFactory* factory)
     : KDecoration(bridge, factory)
 {
+    Q_ASSERT(dynamic_cast< KDecorationBridgeUnstable* >(bridge));
 }
 
 KDecorationUnstable::~KDecorationUnstable()
