--- kde-workspace-4.9.98/powerdevil/kcmodule/activities/powerdevilactivitiesconfig.desktop.0004~	2013-01-15 17:10:50.000000000 +0100
+++ kde-workspace-4.9.98/powerdevil/kcmodule/activities/powerdevilactivitiesconfig.desktop	2013-01-29 02:09:12.707089141 +0100
@@ -39,7 +39,7 @@ Name[pl]=Ustawienia działań
 Name[pt]=Configuração da Actividade
 Name[pt_BR]=Configurações da atividade
 Name[ro]=Configurare activități
-Name[ru]=Настройка активных действий
+Name[ru]=Настройка для комнат
 Name[sk]=Nastavenie aktivity
 Name[sl]=Nastavitve dejavnosti
 Name[sr]=Поставке активности
@@ -81,7 +81,7 @@ Comment[pl]=Konfiguracja zarządzania en
 Comment[pt]=Configurar a Gestão de Energia por Actividade
 Comment[pt_BR]=Configura o gerenciamento de energia por atividade
 Comment[ro]=Configurează gestiunea alimentării per activitate
-Comment[ru]=Настройки параметров энергосбережения для активных действий
+Comment[ru]=Настройка параметров энергосбережения отдельно для каждой комнаты
 Comment[sk]=Nastaviť nastavenia správy napájania podľa aktivity
 Comment[sl]=Nastavitve upravljanja z energijo glede na dejavnost
 Comment[sr]=Подешавање управљања напајањем према активности
@@ -120,7 +120,7 @@ X-KDE-Keywords[nl]=systeem,energie,energ
 X-KDE-Keywords[pl]=system,moc,zarządzanie energią,energia,laptop,bateria,wstrzymanie,AC,wstrzymaj,hibernuj,jasność,wydajność,pokrywka,działanie,działania
 X-KDE-Keywords[pt]=sistema,energia,gestão de energia,portátil,bateria,suspensão,AC,suspender,hibernar,brilho,performance,tampo,actividade,actividades
 X-KDE-Keywords[pt_BR]=sistema,energia,gerenciamento de energia,energia,portátil,laptop,notebook,bateria,suspensão,AC,suspender,hibernar,brilho,desempenho,tampa,atividade,atividades
-X-KDE-Keywords[ru]=system,power,power management,energy,laptop,battery,suspension,AC,suspend,hibernate,brightness,performance,lid,activity,activities,системное,питание,управление питанием,энергопотребление,ноутбук,батарея,спящий режим,гибернация,ждущий режим,яркость,производительность,крышка,комната,комнаты
+X-KDE-Keywords[ru]=system,power,power management,energy,laptop,battery,suspension,AC,suspend,hibernate,brightness,performance,lid,система,питание,электропитание,управление,перейти в спящий режим,спящий режим,яркость,производительность, электропитанием,энергия,ноутбук,лэптоп,батарея,приостановка,приостановить,крышка
 X-KDE-Keywords[sk]=systém,napájanie,správa napájanie,energia,laptop,batéria,uspanie,AC,uspať,hibernovať,jas,výkon,kryt,aktivita,aktivity
 X-KDE-Keywords[sl]=sistem,energija,upravljanje z energijo,prenosnik,prenosni računalnik,baterija,akumulator,pripravljenost,mirovanje,električno omrežje,svetlost,zmogljivost,pokrov,dejavnost,dejavnosti,napajanje
 X-KDE-Keywords[sr]=system,power,power management,energy,laptop,battery,suspension,AC,suspend,hibernate,brightness,performance,lid,activity,activities,систем,напајање,струја,управљање напајањем,лаптоп,батерија,суспендовање,АЦ,хибернација,светлина,осветљај,перформансе,поклопац,активност
--- kde-workspace-4.9.98/powerdevil/kcmodule/global/powerdevilglobalconfig.desktop.0004~	2013-01-29 02:01:31.119529861 +0100
+++ kde-workspace-4.9.98/powerdevil/kcmodule/global/powerdevilglobalconfig.desktop	2013-01-29 02:09:50.963636201 +0100
@@ -82,7 +82,7 @@ Comment[pl]=Konfiguracja zaawansowanych
 Comment[pt]=Configurar a Gestão de Energia Avançada
 Comment[pt_BR]=Configura as opções avançadas de gerenciamento de energia
 Comment[ro]=Configurează opțiunile avansate de gestiune a alimentării
-Comment[ru]=Настройка дополнительных параметров энергосбережения
+Comment[ru]=Дополнительные параметры управления питанием
 Comment[sk]=Nastaviť nastavenia pokročilej správy napájania
 Comment[sl]=Napredne nastavitve upravljanja z energijo
 Comment[sr]=Подешавање напредних поставки за управљање напајањем
