diff -urN kde-workspace-4.11.0/powerdevil/daemon/BackendConfig.cmake kde-workspace-4.11.0-patched/powerdevil/daemon/BackendConfig.cmake
--- kde-workspace-4.11.0/powerdevil/daemon/BackendConfig.cmake	2013-07-10 09:44:32.000000000 +1100
+++ kde-workspace-4.11.0-patched/powerdevil/daemon/BackendConfig.cmake	2013-08-15 18:19:46.813876181 +1100
@@ -45,18 +45,7 @@
     kde4_install_auth_actions(org.kde.powerdevil.backlighthelper ${CMAKE_CURRENT_SOURCE_DIR}/backends/upower/backlight_helper_actions.actions)
 endif (UDEV_FOUND)
 
-########################## HAL Backend #####################################
-
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/backends/hal)
-
-set(powerdevilhalbackend_SRCS
-    backends/hal/halsuspendjob.cpp
-    backends/hal/powerdevilhalbackend.cpp
-)
-
-set(powerdevilhalbackend_LIBS ${KDE4_SOLID_LIBS})
-
 ########################## Daemon variables ################################
 
-set(POWERDEVIL_BACKEND_SRCS ${powerdevilupowerbackend_SRCS} ${powerdevilhalbackend_SRCS})
-set(POWERDEVIL_BACKEND_LIBS ${powerdevilupowerbackend_LIBS} ${powerdevilhalbackend_LIBS})
+set(POWERDEVIL_BACKEND_SRCS ${powerdevilupowerbackend_SRCS})
+set(POWERDEVIL_BACKEND_LIBS ${powerdevilupowerbackend_LIBS})
diff -urN kde-workspace-4.11.0/powerdevil/daemon/powerdevilbackendloader.cpp kde-workspace-4.11.0-patched/powerdevil/daemon/powerdevilbackendloader.cpp
--- kde-workspace-4.11.0/powerdevil/daemon/powerdevilbackendloader.cpp	2013-07-10 09:44:32.000000000 +1100
+++ kde-workspace-4.11.0-patched/powerdevil/daemon/powerdevilbackendloader.cpp	2013-08-15 18:19:46.814876189 +1100
@@ -24,7 +24,6 @@
 #ifdef HAVE_UDEV
 #include "backends/upower/powerdevilupowerbackend.h"
 #endif
-#include "backends/hal/powerdevilhalbackend.h"
 
 #include <KDebug>
 
@@ -44,15 +43,6 @@
     kDebug() << "Failed!";
 #endif
 
-    // If we are here, try HAL
-    kDebug() << "Loading HAL backend...";
-    if (PowerDevilHALBackend::isAvailable()) {
-        kDebug() << "Success!";
-        return new PowerDevilHALBackend(parent);
-    }
-
-    // Fail...
-    kDebug() << "Failed!";
     return 0;
 }
 
